### PR TITLE
chore: fix mac os ci

### PR DESCRIPTION
**Pull request by romange**
Update the runner to macos 13 following this issue https://github.com/actions/runner-images/issues/9997

### CODE REVIEW

File: cmake/third_party.cmake [hunk 1]
```diff
@@ -372,7 +375,7 @@ endif()
 
 add_third_party(
   cares
-  URL https://github.com/c-ares/c-ares/releases/download/cares-1_29_0/c-ares-1.29.0.tar.gz
+  URL https://codeload.github.com/c-ares/c-ares/tar.gz/refs/tags/v1.31.0
```
Comment (romange) [inline cmake/third_party.cmake:378]: update the version (1.29 fails building on macos)

**Review by chakaz — DISMISSED** [review]
:shrug:

**Review by chakaz — APPROVED** [review]

**Comment by romange** [comment]
Sorry, was a bit harder than I thought.

**Comment by chakaz** [comment]
> Sorry, was a bit harder than I thought.

I think that MacOS should be the one apologizing :nerd_face:

**Pull request merged**